src/Packages/Passport/Runtime/Scripts/Private/VersionInfo.cs

```diff
@@ -0,0 +1,32 @@
+using UnityEngine;
+using System.Collections.Generic;
+
+namespace Immutable {
+    public class VersionInfo
+    {
+        public const string Engine = "unity";
+        public static string UnityVersion { get; private set; }
+        public static RuntimePlatform Platform { get; private set; }
+        public static string OsVersion { get; private set; }
+
+        static VersionInfo()
+        {
+            UnityVersion = Application.unityVersion;
+            Platform = Application.platform;
+            OsVersion = SystemInfo.operatingSystem.Replace(" ", "_");
+        }
+
+        public Dictionary<string, string> GetVersionInfo()
+        {
+            Dictionary<string, string> versionInfo = new Dictionary<string, string>
```

Comment:
To clarify my previous comment, I suggested creating a model class to hold the version information. Not moving the code into a different class.

```
[JsonObject(NamingStrategyType = typeof(CamelCaseNamingStrategy))]
public class VersionInfo
  {
      public string Engine;
      public string UnityVersion;
      public string Platform;
     public string PlatformVersion;
  }
```

Or keep it as is in the original file but store the magic strings (e.g. `"engine"`, `"engineVersion"`) as constants at the top.

```
private const string KEY_ENGINE = "engine";
private const string KEY_ENGINE_VERSION = "engineVersion";
...
```